src/panels/lovelace/cards/hui-statistic-card.ts

```diff
@@ -0,0 +1,299 @@
+import {
+  css,
+  CSSResultGroup,
+  html,
+  LitElement,
+  PropertyValues,
+  TemplateResult,
+} from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { applyThemesOnElement } from "../../../common/dom/apply_themes_on_element";
+import { fireEvent } from "../../../common/dom/fire_event";
+import { isValidEntityId } from "../../../common/entity/valid_entity_id";
+import { formatNumber } from "../../../common/number/format_number";
+import "../../../components/ha-card";
+import "../../../components/ha-alert";
+import "../../../components/ha-state-icon";
+import {
+  fetchStatistic,
+  getDisplayUnit,
+  getStatisticLabel,
+  getStatisticMetadata,
+  isExternalStatistic,
+  StatisticsMetaData,
+} from "../../../data/recorder";
+import { HomeAssistant } from "../../../types";
+import { computeCardSize } from "../common/compute-card-size";
+import { findEntities } from "../common/find-entities";
+import { hasConfigOrEntityChanged } from "../common/has-changed";
+import { createHeaderFooterElement } from "../create-element/create-header-footer-element";
+import {
+  LovelaceCard,
+  LovelaceCardEditor,
+  LovelaceHeaderFooter,
+} from "../types";
+import { HuiErrorCard } from "./hui-error-card";
+import { EntityCardConfig, StatisticCardConfig } from "./types";
+
+@customElement("hui-statistic-card")
+export class HuiStatisticCard extends LitElement implements LovelaceCard {
+  public static async getConfigElement(): Promise<LovelaceCardEditor> {
+    await import("../editor/config-elements/hui-statistic-card-editor");
+    return document.createElement("hui-statistic-card-editor");
+  }
+
+  public static getStubConfig(
+    hass: HomeAssistant,
+    entities: string[],
+    entitiesFill: string[]
+  ) {
+    const includeDomains = ["sensor"];
+    const maxEntities = 1;
+    const foundEntities = findEntities(
+      hass,
+      maxEntities,
+      entities,
+      entitiesFill,
+      includeDomains
+    );
+
+    return {
+      entity: foundEntities[0] || "",
+      period: { calendar: { period: "month", offset: 0 } },
+    };
+  }
+
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @state() private _config?: StatisticCardConfig;
+
+  @state() private _value?: number | null;
+
+  @state() private _metadata?: StatisticsMetaData;
+
+  @state() private _error?: string;
+
+  private _interval?: number;
+
+  private _footerElement?: HuiErrorCard | LovelaceHeaderFooter;
+
+  public setConfig(config: StatisticCardConfig): void {
+    if (!config.entity) {
+      throw new Error("Entity must be specified");
+    }
+    if (!config.stat_type) {
+      throw new Error("Statistic type must be specified");
+    }
+    if (!config.period) {
+      throw new Error("Period must be specified");
+    }
+    if (
+      config.entity &&
+      !isExternalStatistic(config.entity) &&
```

Comment:
Ah I see that you explicitly check for isExternal here. Why would we block those ?

Reply:
We don't block, we check if valid entity id or external